### PR TITLE
Refactor your code

### PR DESCRIPTION
**Pull request by tomasesquivelgc**
# In this branch
 Together with my partners @Rebelzob and @ivangonzalez224 have refactored some major parts of the code I originally worked on to follow SOLID principles.
 - We have created separate classes responsible for the creation and listing of users and books.
 - Ensured the code follows KISS, YAGNI and DRY principles
 - Refactored main.rb to minimize it's functions.

### CODE REVIEW

File: Project/create_user.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    people << Teacher.new(age, specialization, name: name)
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    person_type = gets.chomp.to_i
```
Comment (Efiamotu-1) [inline Project/create_user.rb]: - [ ] To follow the `Single Responsibility Rule` for best practices all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. see the example code snippet below
- code snippet
```ruby
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    person_type = gets.chomp
```
- [ ] Please kindly update your code base to follow this rule by implementing the above requirement everywhere in your app where you are requesting for user inputs :+1: :sparkles:

File: Project/main.rb [hunk 2]
```diff
@@ -25,10 +13,30 @@ def main
...
-    handle_option(app, option)
+  def handle_option(option)
+    case option
+    when 1 then @app.create_person
+    when 2 then @app.create_book
+    when 3 then @app.create_rental
+    when 4 then @app.list_books
+    when 5 then @app.list_people
+    when 6 then @app.list_rentals
+    when 7 then exit
```
Comment (Efiamotu-1) [inline Project/main.rb]: - [ ] Following the [reference video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) your project isn't following the correct order as shown below please i kindly suggest you make this necessary correction :+1: :sparkles: 
- project requirement
order
- your project
or

- [ ] Please Kindly ensure that your app follows all of the steps as spresented in the [reference video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) before resubmitting :+1: :sparkles:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @tomasesquivelgc & @ivangonzalez224 & @Rebelzob , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Project/main.rb (hunk 2 above)
Comment (ivangonzalez224) [reply Project/main.rb]: Hi @Efiamotu-1 thank you for the review.
Changes done: 
- Corrected the order of the menu options.
- Changed the input requests and inputs validation to the interface file.

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @Team
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Good refactoring :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**